src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1387,15 +1387,21 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
     uct_rc_txqp_purge_outstanding(&iface->super.super, txqp, ep_status, pi, 0);
 
     ucs_assert(ep->dci != UCT_DC_MLX5_EP_NO_DCI);
+    /* Try to return DCI into iface stack */
     uct_dc_mlx5_iface_dci_put(iface, dci_index);
     uct_dc_mlx5_get_arbiter_params(iface, ep, &waitq, &group, &pool_index);
 
-    /* Do not invoke pending requests on a failed endpoint */
+    /* Do not invoke pending requests on a failed endpoint. This call should be
+     * done after uct_dc_mlx5_iface_dci_put because uct_dc_mlx5_iface_dci_put
+     * may schedule group to arbiter */
```

Comment:
Maybe better remove arbiter schedule from dci_put?

Reply:
yep, why not, make it in separate PR (here is bug-fix)?

Reply:
okay